gloom/srv/shopsdk.py

```diff
@@ -0,0 +1,143 @@
+    with httpimport.remote_repo(["utilsbylarsen"], url):
+      import utilsbylarsen
+    with open(str(os.getcwd()) + "/" + "./config.json", "rb") as f:
+      config = json.load(f)
+    if conf["production"] and conf["send_logs"]:
+      utilsbylarsen.setup_log(config["sentry_url"], False)
+    if number == 1:
+      contenttype = "application/" + mime
+    elif number == 2:
+      contenttype = "video/" + mime
+    elif number == 3:
+      contenttype = "audio/" + mime
+    elif number == 4:
+      contenttype = "image/" + mime
+    elif number == 5:
+      contenttype = "text/" + mime
+    elif number == 6:
+      contenttype = "font/" + mime
+    else:
+       utilsbylarsen.log("THREW EXCEPTION IN ENGINE SCRIPT BECAUSE INTEGER IS: " % number, "CRITICAL")
+       raise ExceptionError
+    message = Mail(
+      from_email=reader.readthefromemail(),
+      to_emails=thetoemail,
+      subject=reader.readsubject(),
+      html_content=reader.readhtml()
+    )
+    with open(filetosend, 'rb') as f:
+      data = f.read()
+      f.close()
+    encodedfile = binascii.b2a_base64(bytes(data)).decode()
+    attachedFile = Attachment(
+      FileContent(encodedfile),
+      FileName(filetosend),
+      FileType(contenttype),
+      Disposition('attachment')
+    )
+    message.attachment = attachedFile
+    sg = SendGridAPIClient(config["sendgrid_api_key"])
+    response = sg.send(message)
+    data = int(currentnoofpoints - pointsneeded)
+    pad24 = str(defs.padding())
+    pad48 = str(pad24) + str(pad24)
+    #Above, it sets up padding, sends the email..
+    #And it even calculates the remaining points.
+    returndata = str(pad24) + str(data) + str(pad24) + str(pad48) + str(response.status_code, response.body, response.headers)
+    #The string after the set of 24 pad strings is the remaining points.
+    #The string after the set of 24 pad strings AND the set of 48 pad strings is the sendgrid result data.
+    #This is important when you utilize this module, to obtain both bits of data seperately.
+    #Otherwise, everything will be completely off.
+    return returndata
+class writer():
+  def writehtml():
+    htmldata = '<strong>Check attachments, we attached your purchase!</strong><br><br><br><strong>Sent by 6100m\'s DLC Mail Bot</strong>'
+    savefile = open(str(os.getcwd()) + "/" + 'html.bin', 'wb')
+    pickle.dump(htmldata, savefile)
+    savefile.close()
+    returndata = defs.zero()
+    return returndata
+  def writesubject():
+    subjectdata = 'Thank you for your purchase! Here is the DLC you purchased!'
+    savefile = open(str(os.getcwd()) + "/" + 'subject.bin', 'wb')
+    pickle.dump(subjectdata, savefile)
+    savefile.close()
+    returndata = defs.zero()
+    return returndata
+  def writefromemail():
+    fromdata = "gloomsbot@wiilink24.com"
+    savefile = open(str(os.getcwd()) + "/" + 'from.bin', 'wb')
+    pickle.dump(fromdata, savefile)
+    savefile.close()
+    returndata = defs.zero()
+    return returndata
```

Comment:
It would be preferable to have these in well-defined templates we could do substitution with (say via jinja2 format strings). This would especially help with the email.